File: FrontEnd/car_rental_angular/src/app/modules/admin/services/admin.service.ts
import { HttpClient, HttpHeaders } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { StorageService } from 'src/app/auth/services/storage/storage.service';

const BASIC_URL = ["http://localhost:8080"];

@Injectable({
  providedIn: 'root'
})
export class AdminService {

  constructor(private http: HttpClient) { }

  postCar(carDto:any):Observable<any>{
    return this.http.post(BASIC_URL+"/api/admin/car",carDto,{
       headers:this.createAuthorizationHeader()
    });
  }

  getAllCars():Observable<any>{
    return this.http.get(BASIC_URL + "/api/admin/cars",{
      headers: this.createAuthorizationHeader()
    })
  }

  deleteCar(id:number):Observable<any>{
    return this.http.delete(BASIC_URL + "/api/admin/car/"+id,{
      headers:this.createAuthorizationHeader()
    });
  }

  getCarById(id:number):Observable<any>{
    return this.http.get(BASIC_URL+"/api/admin/car/"+id,{
      headers:this.createAuthorizationHeader()
    });
  }

  updateCar(carId: number,carDto:any) : Observable<any>{
    return this.http.put(BASIC_URL+"/api/admin/car/"+carId,carDto,{
      headers: this.createAuthorizationHeader()
    })
  }

  getCarBookings():Observable<any>{
    return this.http.get(BASIC_URL + "/api/admin/car/bookings",{
      headers: this.createAuthorizationHeader()
    })
  }

  changeBookingStatus(bookingId: number,status:string):Observable<any>{
    return this.http.get(BASIC_URL + `/api/admin/car/booking/${bookingId}/${status}`,{
      headers: this.createAuthorizationHeader()
    })
  }

  searchCar(searchCarDto:any):Observable<any>{
    return this.http.post(BASIC_URL+"/api/admin/car/search",searchCarDto,{
       headers:this.createAuthorizationHeader()
    });
  }

  createAuthorizationHeader():HttpHeaders{
    let authHeaders: HttpHeaders = new HttpHeaders();
    return authHeaders.set(
      'Authorization',
      'Bearer '+StorageService.getToken()
    );
  }


}

File: Backend/Car_Rental_Spring/src/main/java/com/shriram/Car_Rental_Spring/services/admin/AdminService.java
package com.shriram.Car_Rental_Spring.services.admin;

import com.shriram.Car_Rental_Spring.dto.BookACarDto;
import com.shriram.Car_Rental_Spring.dto.CarDto;
import com.shriram.Car_Rental_Spring.dto.CarDtoListDto;
import com.shriram.Car_Rental_Spring.dto.SearchCarDto;

import java.io.IOException;
import java.util.List;

public interface AdminService {

    boolean postCar(CarDto carDto) throws IOException;
    List<CarDto> getAllCars();
    void deleteCar(Long id);
    CarDto getCarById(Long id);
    boolean updateCar(Long carId,CarDto carDto) throws IOException;

    List<BookACarDto> getBookings();

    boolean changeBookingStatus(Long bookingId,String status);

    CarDtoListDto searchCar(SearchCarDto searchCarDto);


}
